SimPEG/Maps.py

```diff
@@ -709,6 +709,51 @@ def adj(v):
     # inverse = deriv
 
 
+class Projection(IdentityMap):
+    """
+        A map to rearrange / select parameters
+
+        :param int nP: number of model parameters
+        :param numpy.array index: indices to select
+    """
+
+    def __init__(self, nP, index, **kwargs):
+        assert isinstance(index, (np.ndarray, slice, list)), (
+            'index must be a np.ndarray or slice, not {}'.format(type(index)))
+        super(Projection, self).__init__(nP=nP, **kwargs)
+
+        if isinstance(index, slice):
+            index = list(range(*index.indices(self.nP)))
+        self.index = index
+        self._shape = nI, nP = len(self.index), self.nP
+
+        # sparse projection matrix
+        self.P = sp.csr_matrix(
+            (np.ones(nI), (range(nI), self.index)), shape=(nI, nP)
+        )
+
+    def _transform(self, m):
+        return m[self.index]
```

Comment:
`assert len(m) == self.nP, "Ensure that the model has {} elements.".format(self.nP)`